cmake/DetectOptions.cmake

```diff
@@ -170,7 +170,35 @@ endif()
 
 set(mpi_find_components C)
 
-# Cuda
+if(ADIOS_USE_Kokkos AND ADIOS_USE_CUDA)
+  message(FATAL_ERROR "ADIOS2_USE_Kokkos is incompatible with ADIOS_USE_CUDA")
+endif()
+
+# Kokkos
+if(ADIOS2_USE_Kokkos)
+  if(ADIOS2_USE_Kokkos STREQUAL AUTO)
+    find_package(Kokkos 3.7 QUIET)
+  else()
+    find_package(Kokkos 3.7 REQUIRED)
+  endif()
+  if(Kokkos_FOUND)
+    set(ADIOS2_HAVE_Kokkos TRUE)
+    if(Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP OR Kokkos_ENABLE_SYCL)
+      if(Kokkos_ENABLE_CUDA)
+        set(ADIOS2_HAVE_Kokkos_CUDA TRUE)
+        find_package(CUDAToolkit REQUIRED)
```

Comment:
```suggestion
```

Maybe we do not need this anymore with the new Kokkos, I did not try it.

Reply:
indeed it is not needed anymore, it's removed